econml/sklearn_extensions.py

```diff
@@ -0,0 +1,975 @@
+
+    max_iter : int, optional
+        The maximum number of iterations
+
+    tol : float, optional
+        The tolerance for the optimization: if the updates are
+        smaller than ``tol``, the optimization code checks the
+        dual gap for optimality and continues until it is smaller
+        than ``tol``.
+
+    warm_start : bool, optional
+        When set to True, reuse the solution of the previous call to fit as
+        initialization, otherwise, just erase the previous solution.
+        See :term:`the Glossary <warm_start>`.
+
+    random_state : int, RandomState instance or None, optional, default None
+        The seed of the pseudo random number generator that selects a random
+        feature to update.  If int, random_state is the seed used by the random
+        number generator; If RandomState instance, random_state is the random
+        number generator; If None, the random number generator is the
+        RandomState instance used by `np.random`. Used when ``selection`` ==
+        'random'.
+
+    selection : str, default 'cyclic'
+        If set to 'random', a random coefficient is updated every iteration
+        rather than looping over features sequentially by default. This
+        (setting to 'random') often leads to significantly faster convergence
+        especially when tol is higher than 1e-4.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,) | (n_targets, n_features)
+        parameter vector (w in the cost function formula)
+
+    intercept_ : float | array, shape (n_targets,)
+        independent term in decision function.
+
+    n_iter_ : int | array-like, shape (n_targets,)
+        number of iterations run by the coordinate descent solver to reach
+        the specified tolerance.
+    """
+
+    def __init__(self, alpha=1.0, fit_intercept=True, normalize=False,
+                 copy_X=True, max_iter=1000, tol=1e-4, warm_start=False,
+                 random_state=None, selection='cyclic'):
+        super(WeightedMultiTaskLasso, self).__init__(
+            alpha=alpha, fit_intercept=fit_intercept, normalize=False,
+            copy_X=copy_X, max_iter=max_iter, tol=tol, warm_start=warm_start,
+            random_state=random_state, selection=selection)
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit model with coordinate descent.
+
+        Parameters
+        ----------
+        X : ndarray or scipy.sparse matrix, (n_samples, n_features)
+            Data
+
+        y : ndarray, shape (n_samples,) or (n_samples, n_targets)
+            Target. Will be cast to X's dtype if necessary
+
+        sample_weight : numpy array of shape [n_samples]
+                        Individual weights for each sample.
+                        The weights will be normalized internally.
+        """
+        _fit_weighted_linear_model(self, WeightedMultiTaskLasso, X, y, sample_weight)
+        return self
+
+
+class WeightedKFold(object):
```

Comment:
We more frequently do

```suggestion
class WeightedKFold:
```